src/agent/src/image_rpc.rs

```diff
@@ -47,6 +77,213 @@ impl ImageService {
         }
     }
 
+    fn build_oci_path(cid: &str) -> PathBuf {
+        let mut oci_path = PathBuf::from("/tmp");
+        oci_path.push(cid);
+        oci_path.push(IMAGE_OCI);
+        oci_path
+    }
+
+    fn decrypt_layer_data(
+        layer: &ImageLayer,
+        layer_digest: &str,
+        image_manifest: &mut OciManifest,
+        crypto_config: &CryptoConfig,
+        oci_blob_path: &Path,
+    ) -> Result<()> {
+        if let Some(decrypt_config) = &crypto_config.decrypt_config {
+            for layer_desc in image_manifest.layers.iter_mut() {
+                if layer_desc.digest.as_str() == layer_digest {
+                    let (layer_decryptor, _dec_digest) =
+                        decrypt_layer(decrypt_config, layer.data.as_slice(), layer_desc, false)?;
+                    let mut plaintxt_data: Vec<u8> = Vec::new();
+                    let mut decryptor = layer_decryptor.unwrap();
```

Comment:
Please don't use `unwrap()` or `expect()`, except in the scenarios shown in the table in this document:

- https://github.com/kata-containers/kata-containers/blob/main/docs/code-pr-advice.md#rust-language-specifics

Instead, this line should be:

```suggestion
                    let mut decryptor = layer_decryptor?;
```

Reply:
I agree we should not unwrap, however the fix should be:

```rust
let mut decryptor = layer_decryptor.ok_or_else(|| anyhow!("Missing layer decryptor"))?;                                                                                                                                                                                             
```

Reply:
Thanks, fixed now.